From 24d2c9288f2c9da23b9276291947c3423d9e6104 Mon Sep 17 00:00:00 2001
From: Erik Sundin <erik.sundin@freespee.com>
Date: Fri, 5 Apr 2019 12:07:22 +0200
Subject: Remove activation/deactivation of audio session from baresip

---
 modules/audiounit/sess.c | 67 ----------------------------------------
 1 file changed, 67 deletions(-)

diff --git a/modules/audiounit/sess.c b/modules/audiounit/sess.c
index abc966e..e094792 100644
--- a/modules/audiounit/sess.c
+++ b/modules/audiounit/sess.c
@@ -25,34 +25,6 @@ struct audiosess_st {
 
 static struct audiosess *gas;
 
-
-#if TARGET_OS_IPHONE
-static void propListener(void *inClientData, AudioSessionPropertyID inID,
-			 UInt32 inDataSize, const void *inData)
-{
-	struct audiosess *sess = inClientData;
-	CFDictionaryRef dref = inData;
-	CFNumberRef nref;
-	SInt32 reason = 0;
-
-	(void)inDataSize;
-	(void)sess;
-
-	if (kAudioSessionProperty_AudioRouteChange != inID)
-		return;
-
-	nref = CFDictionaryGetValue(
-			dref,
-			CFSTR(kAudioSession_AudioRouteChangeKey_Reason)
-			);
-
-	CFNumberGetValue(nref, kCFNumberSInt32Type, &reason);
-
-	info("audiounit: AudioRouteChange - reason %d\n", reason);
-}
-#endif
-
-
 static void sess_destructor(void *arg)
 {
 	struct audiosess_st *st = arg;
@@ -65,12 +37,6 @@ static void sess_destructor(void *arg)
 static void destructor(void *arg)
 {
 	struct audiosess *as = arg;
-#if TARGET_OS_IPHONE
-	AudioSessionPropertyID id = kAudioSessionProperty_AudioRouteChange;
-
-	AudioSessionRemovePropertyListenerWithUserData(id, propListener, as);
-	AudioSessionSetActive(false);
-#endif
 
 	list_flush(&as->sessl);
 
@@ -85,26 +51,10 @@ int audiosess_alloc(struct audiosess_st **stp,
 	struct audiosess *as = NULL;
 	int err = 0;
 	bool created = false;
-#if TARGET_OS_IPHONE
-	AudioSessionPropertyID id = kAudioSessionProperty_AudioRouteChange;
-	UInt32 category;
-	OSStatus ret;
-#endif
 
 	if (!stp)
 		return EINVAL;
 
-#if TARGET_OS_IPHONE
-	/* Must be done for all modules */
-	category = kAudioSessionCategory_PlayAndRecord;
-	ret = AudioSessionSetProperty(kAudioSessionProperty_AudioCategory,
-				      sizeof(category), &category);
-	if (ret) {
-		warning("audiounit: Audio Category: %d\n", ret);
-		return EINVAL;
-	}
-#endif
-
 	if (gas)
 		goto makesess;
 
@@ -112,23 +62,6 @@ int audiosess_alloc(struct audiosess_st **stp,
 	if (!as)
 		return ENOMEM;
 
-#if TARGET_OS_IPHONE
-	ret = AudioSessionSetActive(true);
-	if (ret) {
-		warning("audiounit: AudioSessionSetActive: %d\n", ret);
-		err = ENOSYS;
-		goto out;
-	}
-
-	ret = AudioSessionAddPropertyListener(id, propListener, as);
-	if (ret) {
-		warning("audiounit: AudioSessionAddPropertyListener: %d\n",
-			ret);
-		err = EINVAL;
-		goto out;
-	}
-#endif
-
 	gas = as;
 	created = true;
 
-- 
2.18.0

